webhook/aws/sns_handler.go

```diff
@@ -402,7 +402,8 @@ func (ss *SNSServer) ListSubscriptionsByMatchingEndpoint() (*list.List, error) {
 		for _, sub := range resp.Subscriptions {
 			timestamp = 0
 			if !strings.EqualFold(*sub.Endpoint, ss.SelfUrl.String()) &&
-				strings.Contains(*sub.Endpoint, endpoint) {
+				strings.Contains(*sub.Endpoint, endpoint) &&
+				strings.Contains(*sub.SubscriptionArn, ss.Config.Sns.TopicArn) {
```

Comment:
Any sample example data to prove this condition?

Reply:
Unfortunately I don't have sample data and don't have the time to recreate the issue on my end. 

I remember we were having an issue with Caduceus confirming its subscriptions, which caused us to have multiple "Pending confirmation" subs on the AWS SNS console for that particular topic. These would be included when requesting current subscriptions. To recreate you should first send some subscription events without confirming them, then restarting the server.